src/device/usbd.c

```diff
@@ -597,6 +597,16 @@ void tud_task_ext(uint32_t timeout_ms, bool in_isr)
       break;
 
       case DCD_EVENT_SOF:
+        TU_LOG_USBD("\r\n");
+        for (uint8_t i = 0; i < TOTAL_DRIVER_COUNT; i++)
+        {
+          usbd_class_driver_t const * driver = get_driver(i);
+          if (driver && driver->sof)
+          {
+            driver->sof(event.rhport, event.sof.frame_count);
+          }
+        }
+        break;
```

Comment:
Here the callback should be executed but instead the ISR function is executed a second time.